lib/dma.h

```diff
@@ -184,11 +184,18 @@ _dma_mark_dirty(const dma_controller_t *dma, const dma_memory_region_t *region,
     }
 }
 
-static inline void
+static inline int
 dma_init_sg(const dma_controller_t *dma, dma_sg_t *sg, dma_addr_t dma_addr,
             uint32_t len, int prot, int region_index)
 {
     const dma_memory_region_t *const region = &dma->regions[region_index];
+    uint32_t reg_prot = region->prot;
+
+    if ((reg_prot == PROT_NONE) ||
```

Comment:
PROT_NONE means 'Pages may not be accessed'. If we return back the sg_list and user tries to touch it, then it will be a segfault. I think user is calling this API in order to access the region so instead better to return failure.

Reply:
Correct, however this function doesn't touch pages, it just translates the DMA address.

Reply:
This sounds like a usage error. Plus, this function is about translating DMA addresses to local addresses, `prot` signifies the intention, it doesn't touch any memory.